app/views/handlers/individual_response.py

```diff
@@ -574,7 +575,19 @@ def handle_get(self):
     def handle_post(self):
         if self.selected_option == self.confirm_option:
             self._update_section_status(CompletionStatus.INDIVIDUAL_RESPONSE_REQUESTED)
-            self._publish_fulfilment_request()
+            try:
+                self._publish_fulfilment_request()
+            except PublicationFailed:
+                retry_url = url_for(
+                    "individual_response.individual_response_post_address_confirm",
+                    list_item_id=self._list_item_id,
+                    **self._request_args,
+                )
+                return (
+                    render_template("errors/individual_response", retry_url=retry_url),
+                    500,
+                )
```

Comment:
If the try except was moved into `_publish_fulfilment_request` you could avoid the duplication in the `handle_post` for mobile and post. You can work out the retry URL from the fact that you have a mobile number etc. The method name might then be more appropriate as `handle_fulfiilment_request` or similar.

Reply:
not sure about `handle_fulfiilment_request` - I think I prefer `handle_fulfilment_request` 😛

Reply:
Actually, this gets uglier when you consider the handle_post then has to check the return value of the fulfilment request method to determine if it has failed or not - in this instance I think it is better to be explicit

Reply:
It doesn't if you are going to redirect to the new page on the exception?

Reply:
This disappears in any case if I follow the approach in #318, as I can handle the logic in the registered error handler.

I'm  making this change now btw, so let me know what you prefer when I make the push

Reply:
Updated in f18f3317 - I've created a new Exception class,  `FulfilmentRequestFailedException`, to handle this apart from generic PubSub PublicationFailed errors.